src/rpc/types.rs

```diff
@@ -183,15 +191,23 @@ impl Rpc {
     // Update the latency of the last n calls.
     // We don't do it within send_request because we might kill it if it times out.
     pub fn update_latency(&mut self, latest: f64) {
+        #[cfg(feature = "prometheusd")]
+        let metric = RpcMetrics::inst(get_storage_registry()).unwrap();  
         // If we have data >= to ma_length, remove the first one in line
         if self.status.latency_data.len() >= self.status.ma_length as usize {
             self.status.latency_data.remove(0);
         }
 
         // Update latency
         self.status.latency_data.push(latest);
-        self.status.latency =
-            self.status.latency_data.iter().sum::<f64>() / self.status.latency_data.len() as f64;
+        let avg = self.status.latency_data.iter().sum::<f64>() / self.status.latency_data.len() as f64;
+        self.status.latency = avg;
+        #[cfg(feature = "prometheusd")]
+        RpcMetrics::push_latency(&metric, &self.url, &self.name, avg);
```

Comment:
i think this is ok. if anything in `push_latency` is blocking, id want it in its own thread and for the data to sent to it for processing by using a channel. using channels might just be cleaner in general and we can avoid using things like statics